lib/improver/tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py

```diff
@@ -0,0 +1,558 @@
+               'radii: 10000.0; lead_times: None; ens_factor: 1.0>')
+        self.assertEqual(result, msg)
+
+
+class Test__find_radii(IrisTest):
+
+    """Test the internal _find_radii function is working correctly."""
+
+    def test_basic_float_cube_lead_times_is_none(self):
+        """Test _find_radii returns a float with the correct value."""
+        neighbourhood_method = CircularNeighbourhood()
+        ens_factor = 0.8
+        num_ens = 2.0
+        radius = 6300
+        plugin = NBHood(neighbourhood_method,
+                        radius,
+                        ens_factor=ens_factor)
+        result = plugin._find_radii(num_ens)
+        expected_result = 3563.8181771801998
+        self.assertIsInstance(result, float)
+        self.assertAlmostEquals(result, expected_result)
+
+    def test_basic_array_cube_lead_times_an_array(self):
+        """Test _find_radii returns an array with the correct values."""
+        neighbourhood_method = CircularNeighbourhood
+        ens_factor = 0.9
+        num_ens = 2.0
+        fp_points = np.array([2, 3, 4])
+        radii = [10000, 20000, 30000]
+        lead_times = [2, 3, 4]
+        plugin = NBHood(neighbourhood_method(),
+                        radii,
+                        lead_times=lead_times,
+                        ens_factor=ens_factor)
+        result = plugin._find_radii(num_ens,
+                                    cube_lead_times=fp_points)
+        expected_result = np.array([6363.961031, 12727.922061, 19091.883092])
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, expected_result)
+
+    def test_interpolation(self):
+        """Test that interpolation is working as expected in _find_radii."""
+        fp_points = np.array([2, 3, 4])
+        neighbourhood_method = CircularNeighbourhood()
+        ens_factor = 0.8
+        num_ens = 4.0
+        fp_points = np.array([2, 3, 4])
+        radii = [10000, 30000]
+        lead_times = [2, 4]
+        plugin = NBHood(neighbourhood_method,
+                        radii,
+                        lead_times=lead_times,
+                        ens_factor=ens_factor)
+        result = plugin._find_radii(num_ens,
+                                    cube_lead_times=fp_points)
+        expected_result = np.array([4000., 8000., 12000.])
+        self.assertArrayAlmostEqual(result, expected_result)
+
+
+class Test_process(IrisTest):
+
+    """Tests for the process method of NeighbourhoodProcessing."""
+
+    RADIUS = 6300  # Gives 3 grid cells worth.
+
+    def setUp(self):
+        """Set up cube."""
+        self.cube = set_up_cube()
+
+    def test_basic(self):
+        """Test that the plugin returns an iris.cube.Cube."""
+        neighbourhood_method = CircularNeighbourhood()
+        result = NBHood(neighbourhood_method, self.RADIUS).process(self.cube)
+        self.assertIsInstance(result, Cube)
+
+    def test_neighbourhood_method_does_not_exist(self):
```

Comment:
New test

Reply:
Maybe we should be raising a better error for this, as I'm not sure I would understand what has gone wrong from the error it raises at the moment. Mayb we need to do 
`try:
     cube_new = self.neighbourhood_method.run(cube_realization,
                                                         radius)
except AttributeError`
etc
In ilne 350 of nbhood.py.

Reply:
Improved error message.